File: app/models/task.rb
class Task < ApplicationRecord
  has_many :comments, dependent: :destroy
  belongs_to :user
  has_and_belongs_to_many :team

  enum status: %i[ongoing finished]

  validates :title, presence: { message: "Can't be empty" }

  def author_name
    user.name + ' ' + user.last_name
  end
end

File: app/javascript/interfaces/task.ts
export type StatusType = "ongoing" | "finished";

export type Task = {
  id: number;
  user_id: number;
  team_id: number;
  title: string;
  body: string;
  created_at: Date;
  update_at: Date;
  status: StatusType;
  author_name: string;
  comments: Comments[];
};

export type Comments = {
  id: number;
  task_id: number;
  body: string;
  created_at: Date;
  update_at: Date;
  user_id: number;
};
